### PR TITLE
Dataset creation docs

### PR DESCRIPTION
**Pull request by JamesOwers**
* Documentation for dataset creation
* Added Table of Contents to first docs
* minor bugfixes to ./make_dataset.py

### CODE REVIEW

File: docs/02_dataset_creation.ipynb [hunk 1]
```diff
@@ -1,11 +1,224 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This call again works with the small brahms data and:\n",
+    "* leaves 44% of the data clean (no degradation is applied)\n",
+    "* selects only `pitch_shift` and `time_shift` degradations\n",
+    "* attempts to perform these degradations at a ratio of 4 to 1 (sampling is done)\n",
+    "* sets some paramters for the `pitch_shift` degradation"
```
Comment (apmcleod) [inline docs/02_dataset_creation.ipynb:103]: Typo here (paramters)

**Review by apmcleod — CHANGES_REQUESTED** [review]
One small typo to correct. I also have 2 slightly larger comments:
1. Do we want to mention "--clean" here? The place would be just after you mention using data from the download cache, just as a small aside, like "(to clear the cache, use the --clean argument)".
2. I am wondering about --degradation-kwargs. Passing in json via command line is extremely clunky with all of the quotes, and I wouldn't recommend using it this way. For the basic tutorial, I would either leave this out completely or say that you can pass kwargs via a json file (or optionally this raw json string). But the json string shouldn't be the "default" way that people will do this, in my opinion.

File: make_dataset.py [hunk 2]
```diff
@@ -375,7 +380,24 @@ def parse_args(args_input=None):
...
-        shutil.rmtree(ARGS.output_dir)
+
+        response = None
+        if not ARGS.no_prompt:
+            response = input(f"Delete output_dir ({ARGS.output_dir})? [y/N]: ")
+
+        if (ARGS.no_prompt) or response in ["y", "ye", "yes"]:
+            try:
+                shutil.rmtree(ARGS.output_dir)
+            except FileNotFoundError:
```
Comment (apmcleod) [inline make_dataset.py:391]: I think this (and above) should be Exception still. rmtree might also raise OSError or others (bad file permissions, missing sym-link, ...)

**Review by apmcleod — CHANGES_REQUESTED** [review]
Small change to 2 except blacks. Otherwise LGTM

File: make_dataset.py (hunk 2 above)
Comment (JamesOwers) [reply make_dataset.py:391]: ok, deal!

**Pull request merged**